card.ml

```diff
@@ -0,0 +1,60 @@
+open Yojson.Basic.Util
+
+(**************************************************************************)
+type color =
+  | R
+  | G
+  | B
+  | Y
+  | ANY
+
+type actions = {
+  skip : bool;
+  reverse : bool;
+  swap : bool;
+  change_color : bool;
+}
+
+type draw_penalty = int
+
+type t = {
+  color : color;
+  actions : actions;
+  penalty : draw_penalty;
+}
+
+(**************************************************************************)
+let j json = Yojson.Basic.from_file json
+
+let to_color s =
+  match String.lowercase_ascii s with
+  | "red" -> R
+  | "green" -> G
+  | "blue" -> B
+  | "yellow" -> Y
+  | "any" -> ANY
+  | _ -> raise (Invalid_argument s)
+
+let to_actions j =
+  {
+    skip = j |> member "skip" |> to_bool;
+    reverse = j |> member "reverse" |> to_bool;
+    swap = j |> member "swap" |> to_bool;
+    change_color = j |> member "change color" |> to_bool;
+  }
```

Comment:
Remember when you are creating private functions, you need to insert specifications.

Reply:
I'll add them right now